server/bundler/utils.js

```diff
@@ -14,6 +14,9 @@ function getAssets( stats ) {
 }
 
 function pathToRegExp( path ) {
+	if ( path === '/' ) {
+		return path;
+	}
 	return new RegExp( '^' + path + '(/.*)?$' );
```

Comment:
@gwwar I agree we should have explicit declarations. I feel relying on some hidden magic that makes wildcard sub-routes match is not intuitive. The problem is that if we remove this right now all routes that are sub-routes would stop working. That includes all cases like: `/posts/:site`.

@dmsnell also completely agree, but that means changing all existing route definitions in `wordpress-com` and making sure they are kept in sync between the section definition and the route handling.

The `/` path check is something we were already doing in the client bundle loader, so this just matches that expectation at least and fixed the current bug. My vote is to merge the simple fix and move towards a better definition of routes.